src/admin_extra/service_admin_extra_migration.erl

```diff
@@ -0,0 +1,788 @@
+    {ok, Data} ->
+      ?MODULE:(list_to_existing_atom("migrate_" ++ binary_to_list(Act)))(Data);
+    Error ->
+      Error
+  end.
+
+%% -------------------------------------------------------------------
+%% @private
+%% @doc
+%% Helper for migrate "pubsub_node" from Mnesia to SQL DB
+%% @end
+%% -------------------------------------------------------------------
+-spec migrate_pubsub_nodes(Data :: list()) -> Result :: tuple().
+
+migrate_pubsub_nodes([]) ->
+  {ok, "Completed the migration of the table 'pubsub_nodes'"};
+
+migrate_pubsub_nodes([H|T]) ->
+  {Host, ID} = H#pubsub_node.nodeid,
+  Owners = jiffy:encode([jid:to_binary(O) || O <- H#pubsub_node.owners]),
+  Ops = jiffy:encode({H#pubsub_node.options}),
+  Cols = ["nidx", "p_key", "name", "type", "owners", "options"],
+  Vals = [H#pubsub_node.id, Host, ID, H#pubsub_node.type, Owners, Ops],
+  Q = ["INSERT INTO pubsub_nodes ", expand_sql_vals(Cols, Vals), ";"],
+  case mongoose_rdbms:sql_query(?HOST, Q) of
+    {error, Reason} ->
+      io:format("The SQL for 'pubsub_node' has error: ~p for values ~p~n", [Reason, Vals]),
+      migrate_pubsub_nodes(T);
+    _ ->
+      migrate_pubsub_nodes(T)
+  end.
+
+%% -------------------------------------------------------------------
+%% @private
+%% @doc
+%% Helper for migrate "pubsub_affiliations" from Mnesia to SQL DB
+%% @end
+%% -------------------------------------------------------------------
+-spec migrate_pubsub_affiliations(Data :: list()) -> Result :: tuple().
+
+migrate_pubsub_affiliations([]) ->
+  {ok, "Completed the migration of the table 'pubsub_affiliations'"};
+
+migrate_pubsub_affiliations([H|T]) ->
+  {SJID, Nidx} = H#pubsub_state.stateid,
+  JidBIn = jid:to_binary(SJID),
+  JID = jid:from_binary(JidBIn),
+  case {JID#jid.luser, H#pubsub_state.affiliation} of
+    {Luser, Aff} when Aff =:= none; Luser =:= <<>> ->
+      migrate_pubsub_affiliations(T);
+    _ ->
+      JidBIn = jid:to_binary(SJID),
+      JID = jid:from_binary(JidBIn),
+      Cols = ["nidx", "luser", "lserver", "aff"],
+      Vals = [Nidx, JID#jid.luser, JID#jid.lserver, convert_aff(H#pubsub_state.affiliation)],
+      Q = ["INSERT INTO pubsub_affiliations ", expand_sql_vals(Cols, Vals), ";"],
+      case mongoose_rdbms:sql_query(?HOST, Q) of
+        {error, Reason} ->
+          io:format("The SQL for 'pubsub_affiliations' has error: ~p for values ~p~n", [Reason, Vals]),
+          migrate_pubsub_affiliations(T);
+        _ ->
+          migrate_pubsub_affiliations(T)
+      end
+  end.
+
+%% -------------------------------------------------------------------
+%% @private
+%% @doc
+%% Helper for migrate "pubsub_subscriptions" from Mnesia to SQL DB
+%% @end
+%% -------------------------------------------------------------------
+-spec migrate_pubsub_subscriptions(Data :: list()) -> Result :: tuple().
+
+migrate_pubsub_subscriptions([]) ->
+  {ok, "Completed the migration of the table 'pubsub_subscriptions'"};
+
+migrate_pubsub_subscriptions([H|T]) ->
```

Comment:
I think it makes sens to migrate from `pubsub_state` to corresponding RDBMS tables in one go over the table.